### PR TITLE
fix: remove availability sets in scaling up template

### LINKED ISSUE #1581: [BUG-0.37.4] Scaling up/down fails with Changing property 'platformFaultDomainCount' is not allowed
**Describe the bug**
- Install kubernetes cluster 1.13.5 with version 0.33.2
- Upgrade to 1.14.3 with version 0.37.4
- Scale up one node
Got this error:
```{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"Conflict","message":"{\r\n \"error\": {\r\n \"code\": \"PropertyChangeNotAllowed\",\r\n \"message\": \"Changing property 'platformFaultDomainCount' is not allowed.\",\r\n \"target\": \"platformFaultDomainCount\"\r\n }\r\n}"}]}```
It seems it's related to this change https://github.com/Azure/aks-engine/pull/1509
**Steps To Reproduce**
- Install kubernetes cluster 1.13.5 with version 0.33.2
- Upgrade to 1.14.3 with version 0.37.4
- Scale up one node

**Expected behavior**
Be able to scale up/down nodes with version 0.37.4

**AKS Engine version**
0.37.4
**Kubernetes version**

1.14.3 or 1.13.7

### PR DESCRIPTION
**Pull request by yangl900**
**Reason for Change**:

The current scaling up implementation for VMAS simply deploys an ARM template, that contains availabilitySet resource. This is not necessary as no change to availability set needed for scaling up or upgrade.

This fix is to remove availabilitySet for scaling up template.

**Issue Fixed**:

I think this fixes #1581
Fixes #1581 
Closes #1607 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/transform/transform.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ const (
 	vmExtensionType  = "Microsoft.Compute/virtualMachines/extensions"
 	nicResourceType  = "Microsoft.Network/networkInterfaces"
 	vnetResourceType = "Microsoft.Network/virtualNetworks"
+	asResourceType   = "Microsoft.Compute/availabilitySets"
```
Comment (CecileRobertMichon) [inline pkg/engine/transform/transform.go]: the rest of the code abbreviates availability sets as "vmas", let's make this `vmasResourceType` for consistency 

(not a blocking suggestion)

Comment (CecileRobertMichon) [reply pkg/engine/transform/transform.go]: (and asIndexes becomes vmasIndexes)

Comment (yangl900) [reply pkg/engine/transform/transform.go]: updated! actually this change failed some unit tests. I'm looking...

Comment (yangl900) [reply pkg/engine/transform/transform.go]: I think unit tests has been fixed with latest commit.

**Comment by mboersma** [comment]
/hold

I want to test this manually against the scenario in #1581 before we decide to merge it.

**Comment by mboersma** [comment]
/hold cancel

I verified that this fixes #1581.

**Pull request merged**